bip-hashrate-escrows.mediawiki

```diff
@@ -0,0 +1,469 @@
+
+People frequently emphasize that miners should have "as little control" as possible. It is a very safe claim to make, and a very easy sentence to write. Much harder is to determine exactly what this minimum value is, and how to achieve it. Harder still is to untie the knot of who is actually controlling what, in a decentralized, interacting system.
+
+Certainly, miners can not have "zero control" -- for that is the same as to just remove them from the system altogether. Some rules are enforced "on miners by nodes" (such as the infamous blocksize limit); other rules are enforced by nodes but are narrowly-controlled by miners (such as the proof-of-work itself, or the block's timestamp). Thirdly, some rules are enforced by both against each other (such as the rule against including invalid txns or double-spent txns), for mutual benefit.
+
+Some pause should be given, after one considers that the sidechain design goal is literally a piece of software that can do *anything*. Anything includes a great many things, many of which I demonstrate to be undesirable. Bitcoin itself does not allow "anything" -- it allows any person to transact, but, in contrast, it does not permit any person to double-spend. This is because "allowing anyone to do anything" is not viable in a world that contains undesirable interactions (what a libertarian might call "aggression") -- in the case of money, these are theft and counterfeiting.
+
+I have produced a comprehensive quantity of written material [1], presentations [2], etc [3] on exactly what the level of miner-control should be, and why. Specifically, I claim that **miners should be aware of the purpose of the sidechain, and they should reject sidechains which have an unclear purpose or which have a purpose that will lead to decrease in miner-wealth** (where wealth measured explicitly as: the estimated present value of the purchasing power of the blockchain's coinbase txns). I claim that this criterion is necessary because, just Original Bitcoin filters unwanted interactions among different BTC txns, so too much "Sidechain Bitcoin" filter out unwanted interactions among sidechain.
+
+* [1] http://www.truthcoin.info/blog/wise-contracts/
+* [2] https://www.youtube.com/watch?v=xGu0o8HH10U&index=1&list=PLw8-6ARlyVciMH79ZyLOpImsMug3LgNc4
+* [3] http://www.drivechain.info/literature/index.html
+
+Call it a "sidechain non-aggression principle", if you want.
+
+To the best of my knowledge, everyone who *has* reviewed this information as found the arguments to be acceptable. It has, also, changed a few minds (from "unacceptable" to "acceptable").
+
+
+===== ISSUE: "Signing" BTC Txns =====
+
+Currently, we use a process which may be suboptimal. It is that we *literally sign* a txn with a globally and publicly known private key. But this is for convenience purposes -- to easily detect the sidechain's balance. The signature that is produced is not doing anything. This is probably an area of improvement.
```

Comment:
If it doesn't do anything, it shouldn't exist at all...

Reply:
As I said, the process as it stands does do something -- it allows tracking of the sidechain's balance.

Reply:
If you have a better way of doing that, we will change it.

Reply:
I don't see how this allows tracking the balance any more than without it.

Reply:
That could be because you know what to do and we don't.

Currently we track the sidechain's balance by adding a kind of "watching only" address, as I said.

But this is something we plan to revisit and remove as soon as we have more time to look into it.

Reply:
You can improve the design now and leave the improvements to the implementation for later.

Why don't you just use a new opcode like blue matt suggested in the mailing list?
Note that it is orthogonal to putting it all on one output or not.

Reply:
> Why don't you just use a new opcode

We will probably add this option

Reply:
We could use one of the expansion opcodes (like NOP4). It seems suboptimal to use one of them up since we do not require any interaction with the script interpreter though. Instead I think it would be best to make use of an existing opcode and some new block level validation rules. 

@psztorc and I had an idea for this involving OP_TRUE(?) + identifying bytes to determine which outputs will have additional restrictions and which sidechain they belong to. The additional restriction could be that all outputs marked for a particular sidechain must be spent by another transaction created by the miner in the (same?) block with the destination again being the weird OP_TRUE script leaving us with a single output after the block is processed for that sidechain. I obviously haven't really figured it out yet! It would be less overhead once figured out though, with no ECDSA operations anymore.